app/forms/housing_costs_form.rb

```diff
@@ -0,0 +1,38 @@
+class HousingCostsForm
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include SessionPersistable
+  include NumberValidatable
+
+  ATTRIBUTES = %i[housing_payments housing_payments_frequency housing_benefit_value housing_benefit_frequency].freeze
+  HOUSING_PAYMENT_FREQUENCIES = %w[every_week every_two_weeks every_four_weeks monthly total].freeze
+
+  attribute :housing_payments, :gbp
+  validates :housing_payments, numericality: { greater_than_or_equal_to: 0, allow_nil: true }, presence: true
+
+  attribute :housing_payments_frequency, :string
+  validates :housing_payments_frequency,
+            presence: true,
+            inclusion: { in: HOUSING_PAYMENT_FREQUENCIES, allow_nil: false },
+            if: -> { send(:housing_payments_frequency).to_i.positive? }
```

Comment:
Here and in other places you don't need to do `send(:housing_payments_frequency)`, as you can simply do `housing_payments_frequency`